lib/supabase.ex

```diff
@@ -166,8 +166,37 @@ defmodule Supabase do
     apply(__MODULE__, which, [])
   end
 
-  @json_library Application.compile_env(:supabase, :json_library, Jason)
+  @json_library Application.compile_env(:supabase_potion, :json_library, Jason)
 
   @doc "Returns the configured JSON encoding library for Supabase libraries."
+  @spec json_library :: Poison | Jason | JSON
   def json_library, do: @json_library
+
+  @doc false
+  defguardp is_atom_opt(atom) when atom in ~w(atoms atoms!)a
+
+  @doc false
+  def decode_json(term, opts) do
+    keys = Keyword.get(opts, :keys)
+
+    with {:ok, decoded} <- json_library().decode(term) do
+      if is_atom_opt(keys), do: atom_keys(decoded, keys), else: decoded
+    end
+  end
+
+  defp atom_keys(term, atom) when is_list(term) do
+    Enum.map(term, &atom_keys(&1, atom))
+  end
+
+  defp atom_keys(term, atom) when is_map(term) do
+    key =
+      if atom == :atoms!,
+        do: &String.to_existing_atom/1,
+        else: &String.to_atom/1
```

Comment:
This does not seem very safe and could be a memory issue if keys change often. This might be the case in RPC calls where the user sets the key, and also in edge function calls. 
Would think that downstream libs switching to string keys would be safer..

Reply:
@Ziinc whay do you mean by dynamic keys?

i've kinda replicated the a behavior  of json libraries like Jason and Poison to have an option to automatic convert keys to atoms

i thought of it because the native JSON module doesn't have this option, so to build an unified api i've replicated the other two options 

but thinking better, it makes more sense to handle only strong keys (which is actually partially done on downstream libraries) and remove that possible memory leak, letting users to handle the key conversion by their own

Reply:
For example if a response returns a uuid as a json key. That would be one example.

Having the option is fine, but it shouldn't be a default and the user should explicitly set it knowing the dangers of it.

Reply:
understood, but this option shouldn't be used to final users directly, only downstream libraries

so given our points, i think the happy path is to simply remove it and deal only with string keys